app/src/main/java/com/vero/tasky/agenda/domain/repository/TaskRepository.kt

```diff
@@ -0,0 +1,19 @@
+package com.vero.tasky.agenda.domain.repository
+
+import com.vero.tasky.agenda.domain.model.AgendaItem
+import com.vero.tasky.agenda.domain.model.ModificationType
+import kotlinx.coroutines.flow.Flow
+
+interface TaskRepository {
+
+    suspend fun saveTask(
+        task: AgendaItem.Task,
+        modificationType: ModificationType
+    ) : Result<Unit>
+
+    fun getTask(id: String): Flow<AgendaItem.Task>
```

Comment:
No need for flow if you arent going to observe for updates (explained on another comment)